File: LeetCode/Integer To Roman/Integer_To_Roman.cpp
#include<bits/stdc++.h>

using namespace std;

// Function to convert the number to roman numeral
string intToRoman(int number) {

    int num[] = {1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000};
    // Possible Roman Alphabets
    string sym[] = {"I", "IV", "V", "IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M"};
    int i = 12;
    string ans = "";

    while(number) {
        int div = number/num[i];
        number = number%num[i];
        while(div--) {
            ans += sym[i];
        }
        i--;
    }

    return ans;
}

// Driver Code
int main() {

    int N;
    //N = 50;
    cout << "Enter the number: ";
    cin >> N;
    cout << intToRoman(N) << "\n";

    return 0;
}

File: LeetCode/Integer To Roman/Integer To Roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        
        numeral_map = {
            1000: 'M',
            900: 'CM',
            500: 'D',
            400: 'CD',
            100: 'C',
            90: 'XC',
            50: 'L',
            40: 'XL',
            10: 'X',
            9: 'IX',
            5: 'V',
            4: 'IV',
            1: 'I'
        }
        
		# Use integer division to get the number of multiples of the value in the number
        # E.g. 994 // 100 == 9
		# Use modulo to get the remainder of num / value
        # E.g. 994 % 100 == 94
        roman = ""
        for value in numeral_map.keys():
            multiplier = num // value
            if multiplier > 0:
                roman += numeral_map[value] * multiplier  # Add the roman numeral multiplyer times
                num %= value
        
        return roman
